Sources/Amplitude/Utilities/HttpClient.swift

```diff
@@ -9,17 +9,19 @@ import Foundation
 
 class HttpClient {
     let configuration: Configuration
-    internal var session: URLSession
+    internal let session: URLSession
 
     init(configuration: Configuration) {
         self.configuration = configuration
-        // shared instance has limitations but think we are not affected
-        // https://developer.apple.com/documentation/foundation/urlsession/1409000-shared
-        self.session = URLSession.shared
+
+        let sessionConfiguration = URLSessionConfiguration.ephemeral
+        sessionConfiguration.httpMaximumConnectionsPerHost = 2
+        self.session = URLSession(configuration: sessionConfiguration)
```

Comment:
I think we can set `delegate=null` here to [prevent memory leaks](https://developer.apple.com/documentation/foundation/nsurlsession#2934758). I noticed [that's what our friends have](https://github.com/segmentio/analytics-swift/blob/main/Sources/Segment/Utilities/HTTPClient.swift#L174-L176).

Reply:
Updated